components/VotePoll.tsx

```diff
@@ -0,0 +1,257 @@
+import React, { Dispatch, SetStateAction } from "react";
+import { useEffect } from "react";
+import { useState } from "react";
+import Button from "@mui/material/Button";
+import Typography from "@mui/material/Typography";
+import { Alert, Snackbar } from "@mui/material";
+import {
+  DragDropContext,
+  Draggable,
+  Droppable,
+  DropResult,
+} from "react-beautiful-dnd";
+import CandidateSelectionBox from "./CandidateSelectionBox";
+import CandidateChoiceBox from "./CandidateChoiceBox";
+import { CreateVote } from "./api";
+import { useFetch } from "../hooks/useFetch";
+
+type Props = {
+  pollData: Poll;
+  setPageAlert: Dispatch<SetStateAction<AlertShape | null>>;
+};
+
+function VotePoll({ pollData, setPageAlert }: Props) {
+  const [_, createVote] = useFetch(CreateVote);
+  const [candidates, setCandidates] = useState<string[]>([]);
+  const [chosenCandidates, setChosenCandidates] = useState<string[]>([]);
+
+  useEffect(() => {
+    if (pollData.candidateList) {
+      setCandidates(pollData.candidateList.map((candidate) => candidate.name));
+    }
+  }, [pollData.candidateList]);
+
+  type DropId = "candidatesDropId" | "chosenCandidatesDropId";
+
+  const dropIdMap = {
+    candidatesDropId: {
+      list: candidates,
+      setList: setCandidates,
+    },
+    chosenCandidatesDropId: {
+      list: chosenCandidates,
+      setList: setChosenCandidates,
+    },
+  };
+
+  const onDragEnd = (result: DropResult) => {
+    const { source, destination } = result;
+    if (!destination) return;
+    const sourceId = source.droppableId as DropId;
+    const destId = destination.droppableId as DropId;
+
+    if (sourceId === destId) {
+      const reOrderedItems = reorder(
```

Comment:
What is the difference between re-order and move?

In this path where `sourceId === destId` you end up calling both but I think the move is potentially doing the same thing as the re-order

Reply:
re-order happens re-orders items for one list, and move moves items from one list to another. move is not called when `sourceId === destId`
let me know if there's anything I can do to make the code clearer